src/core/lib/support/wrap_memcpy.c

```diff
@@ -40,7 +40,7 @@
  */
 
 #ifdef __linux__
-#ifdef __x86_64__
+#if defined(__x86_64__) && defined(__GNU_LIBRARY__)
```

Comment:
I've opened #10560 for tracking.  I'd also want to investigate performance differences in memmove/memcpy before making that change.